storey/drivers.py

```diff
@@ -501,3 +509,450 @@ async def _get_item(self, container, table_path, key, attributes):
+            if update_ok and aggr_item:
+                await self._fetch_state_by_key(aggr_item, container, table_path, key)
+            # else:
+                # should_raise_error = True
+
+        # if should_raise_error:
+        #     raise RedisError(
+        #         f'Failed to save aggregation for {table_path}/{key}. Response status code was {response.status_code}: {response.body}\n'
+        #         f'Update expression was: {update_expression}'
+        #     )
+
+    async def _get_all_fields(self, redis_key: str):
+        try:
+            # TODO: This should be HSCAN, not HGETALL, to avoid blocking Redis
+            # with very large hashes.
+            values = await asyncify(self.redis.hgetall)(redis_key)
+        except redis.ResponseError as e:
+            raise RedisError(f'Failed to get key {redis_key}. Response error was: {e}')
+        res = {self._convert_to_str(key): self._convert_redis_value_to_python_obj(val) for key, val in values.items()
+               if not str(key).startswith(self._aggregation_prefixes)}
+        return res
+
+    async def _get_specific_fields(self, redis_key: str, attributes: List[str]):
+        non_aggregation_attrs = [
+            name for name in attributes
+            if not name.startswith(self._aggregation_prefixes)
+        ]
+        try:
+            values = await asyncify(self.redis.hmget)(redis_key, non_aggregation_attrs)
+        except redis.ResponseError as e:
+            raise RedisError(f'Failed to get key {redis_key}. Response error was: {e}') from e
+        return [{self._convert_to_str(k): self._convert_redis_value_to_python_obj(v)} for k, v in values.items()]
+
+    async def _load_by_key(self, container, table_path, key, attributes):
+        """
+        Return all static attributes, or certain attributes.
+
+        NOTE: Following the V3IO driver's implementation, this method will not
+        return aggregation attributes or associated time values -- just static
+        data, AKA "additional data."
+        """
+        redis_key_prefix = self.make_key(container, table_path, key)
+        static_key = self._static_data_key(redis_key_prefix)
+        if attributes == "*":
+            values = await self._get_all_fields(static_key)
+        else:
+            values = await self._get_specific_fields(static_key, attributes)
+        return values
+
+    async def _get_associated_time_attr(self, redis_key_prefix, aggr_name):
+        aggr_without_relevant_attr = aggr_name[:-2]
+        feature_name_only = aggr_without_relevant_attr[:aggr_without_relevant_attr.rindex('_')]
+        feature_with_relevant_attr = f"{feature_name_only}{aggr_name[-2:]}"
+        associated_time_key = self._aggregation_time_key(redis_key_prefix,
+                                                         feature_with_relevant_attr)
+        time_val = await asyncify(self.redis.get)(associated_time_key)
+        time_val = self._convert_to_str(time_val)
```

Comment:
From what I see, we assign a string to `time_val`, convert it to a string (?) then convert it to an int just below. Could it be that this line is unnecessary?

Reply:
redis returns bstr , not str